azure-iot-device/samples/advanced-hub-scenarios/provision_x509_and_send_telemetry.py

```diff
@@ -0,0 +1,74 @@
+# -------------------------------------------------------------------------
+# Copyright (c) Microsoft Corporation. All rights reserved.
+# Licensed under the MIT License. See License.txt in the project root for
+# license information.
+# --------------------------------------------------------------------------
+import os
+import asyncio
+from azure.iot.device import X509
+from azure.iot.device.aio import ProvisioningDeviceClient
+from azure.iot.device.aio import IoTHubDeviceClient
+from azure.iot.device import Message
+import uuid
+
+
+provisioning_host = os.getenv("PROVISIONING_HOST")
+id_scope = os.getenv("PROVISIONING_IDSCOPE")
+registration_id = os.getenv("DPS_X509_REGISTRATION_ID")
+messages_to_send = 10
+
+
+async def main():
+    x509 = X509(
+        cert_file=os.getenv("X509_CERT_FILE"),
+        key_file=os.getenv("X509_KEY_FILE"),
+        pass_phrase=os.getenv("PASS_PHRASE"),
+    )
+
+    async def register_device():
+        provisioning_device_client = ProvisioningDeviceClient.create_from_x509_certificate(
+            provisioning_host=provisioning_host,
+            registration_id=registration_id,
+            id_scope=id_scope,
+            x509=x509,
+        )
+
+        return await provisioning_device_client.register()
+
+    results = await asyncio.gather(register_device())
+    registration_result = results[0]
+    print("The complete registration result is")
+    print(registration_result.registration_state)
+
+    print("Will send telemetry from the provisioned device")
+    device_client = IoTHubDeviceClient.create_from_registration_result_and_x509(
```

Comment:
>create_from_registration_result_and_x509 [](start = 39, length = 40)

Suggest `create_from_registration_result_and_x509_certificate` to match `create_from_x509_certificate`